anchor/src/instructions.rs

```diff
@@ -0,0 +1,77 @@
+use crate::*;
+use anchor_lang::{prelude::*, solana_program::program};
+
+pub fn add_user_and_deposit<'a, 'b, 'c, 'info>(
+    ctx: CpiContext<'a, 'b, 'c, 'info, AddUserAndDeposit<'info>>,
+    amount: u64,
+    pool_id: u8,
+    page_id: u16,
+) -> Result<()> {
+    let ix = apricot_client::instructions::add_user_and_deposit(
+        ctx.accounts.user_wallet.key,
+        ctx.accounts.user_spl.key,
+        amount,
+        pool_id,
+        page_id,
+    );
+    program::invoke_signed(
+        &ix,
+        &[
+            ctx.program,
```

Comment:
Wouldn't CPI require all accounts for the instructions being invoked? All accounts are available in `add_user_and_deposit_full`. Same comment for all the other anchor instructions.